ext/v8/property-callback.cc

```diff
@@ -74,4 +117,112 @@ namespace rr {
+
+    VALUE rb_property = String(isolate, property->ToString());
+
+    Unlocker unlock(info.GetIsolate());
+    rb_funcall(code, rb_intern("call"), 2, rb_property, (VALUE)PropertyCallbackInfo::Integer(info));
+  }
+
+  void PropertyCallback::invokeNamedPropertyDeleter(v8::Local<v8::String> property, const v8::PropertyCallbackInfo<v8::Boolean>& info) {
+    v8::Isolate* isolate = info.GetIsolate();
+
+    VALUE code(unwrapDeleter(isolate, info.Data()));
+
+    VALUE rb_property = String(isolate, property->ToString());
+
+    Unlocker unlock(info.GetIsolate());
+    rb_funcall(code, rb_intern("call"), 2, rb_property, (VALUE)PropertyCallbackInfo::Boolean(info));
+  }
+
+  void PropertyCallback::invokeNamedPropertyEnumerator(const v8::PropertyCallbackInfo<v8::Array>& info) {
+    v8::Isolate* isolate = info.GetIsolate();
+
+    VALUE code(unwrapEnumerator(isolate, info.Data()));
+
+    Unlocker unlock(info.GetIsolate());
+    rb_funcall(code, rb_intern("call"), 1, (VALUE)PropertyCallbackInfo::Array(info));
+  }
+
+  void PropertyCallback::invokeIndexedPropertyGetter(uint32_t index, const v8::PropertyCallbackInfo<v8::Value>& info) {
+    v8::Isolate* isolate = info.GetIsolate();
+
+    VALUE code(unwrapGetter(isolate, info.Data()));
+
+    Unlocker unlock(info.GetIsolate());
+    rb_funcall(code, rb_intern("call"), 2, UINT2NUM(index), (VALUE)PropertyCallbackInfo::Value(info));
+  }
+
+  void PropertyCallback::invokeIndexedPropertySetter(uint32_t index, v8::Local<v8::Value> value, const v8::PropertyCallbackInfo<v8::Value>& info) {
+    v8::Isolate* isolate = info.GetIsolate();
+
+    VALUE code(unwrapSetter(isolate, info.Data()));
+
+    Unlocker unlock(info.GetIsolate());
+    rb_funcall(
+      code, rb_intern("call"), 3,
+      UINT2NUM(index),
+      (VALUE)Value(isolate, value),
+      (VALUE)PropertyCallbackInfo::Value(info)
+    );
+  }
+
+  void PropertyCallback::invokeIndexedPropertyQuery(uint32_t index, const v8::PropertyCallbackInfo<v8::Integer>& info) {
+    v8::Isolate* isolate = info.GetIsolate();
+
+    VALUE code(unwrapQuery(isolate, info.Data()));
+
+    Unlocker unlock(info.GetIsolate());
+    rb_funcall(code, rb_intern("call"), 2, UINT2NUM(index), (VALUE)PropertyCallbackInfo::Integer(info));
+  }
+
+  void PropertyCallback::invokeIndexedPropertyDeleter(uint32_t index, const v8::PropertyCallbackInfo<v8::Boolean>& info) {
+    v8::Isolate* isolate = info.GetIsolate();
+
+    VALUE code(unwrapDeleter(isolate, info.Data()));
+
+    Unlocker unlock(info.GetIsolate());
+    rb_funcall(code, rb_intern("call"), 2, UINT2NUM(index), (VALUE)PropertyCallbackInfo::Boolean(info));
+  }
+
+  void PropertyCallback::invokeIndexedPropertyEnumerator(const v8::PropertyCallbackInfo<v8::Array>& info) {
+    v8::Isolate* isolate = info.GetIsolate();
+
+    VALUE code(unwrapEnumerator(isolate, info.Data()));
+
+    Unlocker unlock(info.GetIsolate());
+    rb_funcall(code, rb_intern("call"), 1, (VALUE)PropertyCallbackInfo::Array(info));
+  }
+
 }
```

Comment:
I'm not really a fan of this approach because the usage code becomes less clearer. You have to really think about which conversion operators are there on the object you provide, the compiler errors become less clear if you miss declaring something properly and for a new person looking at the code and seeing the same object being passed to 5 different arguments with different types it becomes a lot harder to understand what calls what.

However, I do understand the benefit of not having to do a conditional check for `nil` and that we should be consistent with this and since most of the other code works this way, I will change it.